events/events.go

```diff
@@ -73,11 +73,11 @@ func (em *EventManager) broadcastEvent(evt *XRPCStreamEvent) {
 			select {
 			case s.outgoing <- evt:
 			case <-s.done:
+			default:
+				log.Warnw("event overflow", "bufferSize", len(s.outgoing), "ident", s.ident)
```

Comment:
Might be helpful to add like `dropping slow consumer due to event buffer overflow` here to give the log message a bit more context.